README.md

```diff
@@ -23,67 +23,157 @@ For a Getting started guide, API docs, recipes, making a plugin, etc. see the [d
 This file is just a quick sample to give you a taste of what gulp does.
 
 ```js
-var gulp = require('gulp');
-var coffee = require('gulp-coffee');
-var concat = require('gulp-concat');
-var uglify = require('gulp-uglify');
-var imagemin = require('gulp-imagemin');
-var sourcemaps = require('gulp-sourcemaps');
-var del = require('del');
+var gulp = require('gulp'),
+  rename = require('gulp-rename'),
+  less = require('gulp-less'),
+  cssmin = require('gulp-cssmin'),
+  concat = require('gulp-concat'),
+  uglify = require('gulp-uglify'),
+  del = require('del');
 
 var paths = {
-  scripts: ['client/js/**/*.coffee', '!client/external/**/*.coffee'],
-  images: 'client/img/**/*'
+  styles: {
+    src: 'src/styles/**/*.less',
+    dest: 'assets/styles/'
+  },
+  scripts: {
+    src: 'src/scripts/**/*.js',
+    dest: 'assets/scripts/'
+  }
 };
 
-/* Register some tasks to expose to the cli */
-gulp.task('build', gulp.series(
-  clean,
-  gulp.parallel(scripts, images)
-));
+/* Not all tasks need to use streams, a gulpfile is just another node program
+ * and you can use all packages available on npm, but it must return either a
+ * Promise, a Stream or take a callback and call it
+ */
+function clean() {
+  // You can use multiple globbing patterns as you would with `gulp.src`,
+  // for example if you are using del 2.0 or above, return its promise
+  return del([ 'assets' ]);
+}
+
+/* 
+ * Define our tasks using plain functions
+ */
+function styles() {
+  return gulp.src(paths.styles.src)
+    .pipe(less())
+    .pipe(cssmin())
+    // pass in options to the task
+    .pipe(rename({
+          basename: 'main',
+          suffix: '.min'
+        }))
+    .pipe(gulp.dest(paths.styles.dest));
+}
+
+function scripts() {
+  return gulp.src(paths.scripts.src)
```

Comment:
@contra I'd prefer these to stay named functions instead of arrow functions assigned to a variable.  Registering named functions by the `Function.name` is a killer feature.